lib/spack/spack/ci.py

```diff
@@ -665,6 +672,14 @@ def generate_gitlab_ci_yaml(env, print_summary, output_file, prune_dag=False,
                 pkg_name = pkg_name_from_spec_label(spec_label)
                 release_spec = root_spec[pkg_name]
 
+                # Check if this spec is in our list of known failures.
+                if broken_specs_url:
+                    full_hash = release_spec.full_hash()
+                    broken_spec_path = '{0}/{1}'.format(broken_specs_url, full_hash)
```

Comment:
Taking another look, it occurred to me this is something I would now do with:

```
import spack.util.url as url_util
url_util.join(broken_specs_url, full_hash)
```

It's not fundamentally different from what you've done, and I took a look through the module and found some variety in how I formed urls over time, so you didn't have consistent examples from which to draw.  But the import is already there, there's one other example of using it in the module, and I think it would be good to move toward using one approach to url formatting.

Reply:
👍